src/api/MainSolver.cc

```diff
@@ -115,6 +117,7 @@ void MainSolver::insertFormula(PTRef fla)
     // TODO: Move this to preprocessing of the formulas
     fla = IteHandler(logic, getPartitionManager().getNofPartitions()).rewrite(fla);
 
+    //? TK: is this really related only to interpolants, or to proofs in general?
     if (getConfig().produce_inter()) {
```

Comment:
I think this should call the method `MainSolver::trackPartitions` that I suggest to create below.